File: old_groovy/src/main/groovy/net/zomis/brainf/model/ast/tree/PrintSyntax.java
package net.zomis.brainf.model.ast.tree;

import net.zomis.brainf.model.BrainfuckRunner;

public class PrintSyntax extends Syntax {

    @Override
    public void perform(BrainfuckRunner runner) {
        char write = (char) runner.getMemory().getValue();
        runner.appendOutput(write);
    }

}

File: brainduck-kotlin/src/commonMain/kotlin/net/zomis/brainf/model/ast/tree/PrintSyntax.kt
package net.zomis.brainf.model.ast.tree;

import net.zomis.brainf.model.BrainfuckRunner;

class PrintSyntax : Syntax() {

    override fun perform(runner: BrainfuckRunner) {
        val write = runner.memory.value
        runner.appendOutput(write.toChar());
    }

}
